src/storage/lock_manager/waiter_manager.rs

```diff
@@ -90,7 +94,14 @@ impl WaitTable {
         self.wait_table.iter().map(|(_, v)| v.len()).sum()
     }
 
+    fn flip_wait_table_is_empty(&self, is_empty: bool) {
```

Comment:
How about name it `sync_wait_table_is_empty` and 
```rust
WAIT_TABLE_IS_EMPTY.store(self.wait_table.is_empty(), Ordering::Relaxed);
```

```

Reply:
The logic is flip-flop. If the wait table is empty and added a new waiter, call `flip_wait_table_is_empty(false)`. If the wait table is not empty and removed some waiters, call `flip_wait_table_is_empty(true)`.

Reply:
Only when the wait table becomes empty or becomes non-empty, `WAIT_TABLE_IS_EMPTY` will be set.

Reply:
The logic is too complex.